File: FinanceApp.Shared/Dto/Spending/UpdateSpending.cs
using FinanceApp.Shared.Dto.Base;

namespace FinanceApp.Shared.Dto.Spending
{
    public class UpdateSpending : UpdateDto
    {
        public string Name { get; set; }
        public int Payment { get; set; }
        public int? CreditCardId { get; set; }
        public double Amount { get; set; }
        public int Recurrence { get; set; }
        public bool IsRequired { get; set; }
        public int? CategoryId { get; set; }
        public DateTime InitialDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsEndless { get; set; }
        public int? TimesRecurrence { get; set; }
    }
}

File: FlutterApp/finance_app/lib/models/spending/update_spending.dart
class UpdateSpending {
  String id;
  String name;
  int recurrence;
  String? creditCardId;
  double amount;
  int payment;
  bool isRequired;
  String? categoryId;
  DateTime initialDate;
  DateTime? endDate;
  bool isEndless;
  int timesRecurrence;

  UpdateSpending(
      {required this.id,
      required this.name,
      required this.creditCardId,
      required this.recurrence,
      required this.payment,
      required this.amount,
      required this.isRequired,
      required this.categoryId,
      required this.initialDate,
      required this.endDate,
      required this.isEndless,
      required this.timesRecurrence});

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'recurrence': recurrence,
        'amount': amount,
        'payment': payment,
        'creditCardId': creditCardId,
        'isRequired': isRequired,
        'categoryId': categoryId,
        'initialDate': initialDate.toString(),
        'endDate': endDate?.toString(),
        'isEndless': isEndless,
        'timesRecurrence': timesRecurrence,
      };
}
